synapse/http/matrixfederationclient.py

```diff
@@ -264,14 +264,33 @@ def _request(self, destination, method, path,
             defer.returnValue(response)
 
     def sign_request(self, destination, method, url_bytes, headers_dict,
-                     content=None):
+                     content=None, destination_is=None):
+        """
+        Signs a request by adding an Authorization header to headers_dict
+        Args:
+            destination (str): The desination home server of the request. May be null if the
```

Comment:
I think these are all meant to be `bytes` rather than `str` fwiw. Especially `url_bytes` which you are carefully `.encode`ing

Reply:
point